tests/test_websockets.py

```diff
@@ -0,0 +1,242 @@
+                GUID = "258EAFA5-E914-47DA-95CA-C5AB0DC85B11"
+                key = re.search("sec-websocket-key: ([A-Za-z0-9+/=]*)", decoded, re.IGNORECASE).group(1)
+
+                to_hash = "{:s}{:s}".format(key, GUID)
+                hashed = hashlib.sha1(to_hash.encode("utf8"))
+                encoded = base64.b64encode(hashed.digest()).decode("utf8")
+
+                response_headers["Sec-WebSocket-Accept"] = encoded
+
+                # Respond with the correct hash
+                response = get_websocket_response(response_headers)
+
+                _self.request.sendall(response)
+
+        return WebsocketHandler
+
+    def test_successful_connection(self, proto_ver, proto_name,
+                                   fake_websocket_broker,
+                                   init_response_headers):
+        """ Connect successfully, on correct path """
+
+        mqttc = client.Client(
+            "test_successful_connection",
+            protocol=proto_ver,
+            transport="websockets"
+            )
+
+        response = self._get_callback_handler(init_response_headers)
+
+        with fake_websocket_broker.serve(response):
+            mqttc.connect("localhost", 1888, keepalive=10)
+
+            mqttc.disconnect()
+
+    @pytest.mark.parametrize("mqtt_path", [
+        "/mqtt"
+        "/special",
+        None,
+    ])
+    def test_correct_path(self, proto_ver, proto_name, fake_websocket_broker,
+                          mqtt_path, init_response_headers):
+        """ Make sure it can connect on user specified paths """
+
+        mqttc = client.Client(
+            "test_correct_path",
+            protocol=proto_ver,
+            transport="websockets"
+            )
+
+        mqttc.ws_set_options(
+            path=mqtt_path,
+        )
+
+        def create_response_hash(decoded):
+            # Make sure it connects to the right path
+            assert re.search("GET {:s} HTTP/1.1".format(mqtt_path), decoded, re.IGNORECASE) is not None
+
+        response = self._get_callback_handler(init_response_headers)
+
+        with fake_websocket_broker.serve(response):
+            mqttc.connect("localhost", 1888, keepalive=10)
+
+            mqttc.disconnect()
+
+    @pytest.mark.parametrize("auth_headers", [
+        {"Authorization": "test123"},
+        {"Authorization": "test123", "auth2": "abcdef"},
+        # Won't be checked, but make sure it still works even if the user passes it
+        None,
+    ])
+    def test_correct_auth(self, proto_ver, proto_name, fake_websocket_broker,
+                          auth_headers, init_response_headers):
+        """ Make sure it sends the right auth headers """
+
+        mqttc = client.Client(
+            "test_correct_path",
+            protocol=proto_ver,
+            transport="websockets"
+            )
+
+        mqttc.ws_set_options(
+            headers=auth_headers,
+        )
+
+        def create_response_hash(decoded):
```

Comment:
This method is never used. I think it should be the "check_request" parameter of _get_callback_handler